contracts/allocators/OlympusKP3RHolder.sol

```diff
@@ -0,0 +1,176 @@
+        require(address(authority.vault()) != address(treasury), "No change");
+        treasury = ITreasury(authority.vault());
+    }
+
+    /// @notice             Manages KP3R from treasury and creates lock in vKP3R
+    /// @param _amount      Amount of KP3R that will be managed from treasury and used to create lock
+    /// @param _unlockTime  Timestamp at which lock will be over
+    function createLock(uint256 _amount, uint256 _unlockTime) external onlyGuardian {
+
+        // retrieve amount of KP3R from treasury
+        treasury.manage(address(KP3R), _amount); 
+
+        // approve and deposit into curve
+        KP3R.approve(address(KP3RVault), _amount); 
+
+        KP3RVault.create_lock(_amount, _unlockTime);
+    }
+
+    /// @notice         Manages KP3R from treasury adds to already exisiting lock
+    /// @param _amount  Amount of KP3R that will be managed from treasury and used to add to already existing lock
+    /// @param _manage  Bool if managing KP3R from treasury
+    function increaseAmount(uint256 _amount, bool _manage) external onlyGuardian {
+        if(_manage) { treasury.manage(address(KP3R), _amount); }
+
+        // approve and deposit into curve
+        KP3R.approve(address(KP3RVault), _amount); 
+
+        KP3RVault.increase_amount(_amount);
+    }
+
+    /// @notice             Increases unlock time of existing lock
+    /// @param _unlockTime  Updated Timestamp at which lock will be over
+    function increaseLockTime(uint256 _unlockTime) external onlyGuardian {
+        KP3RVault.increase_unlock_time(_unlockTime);
+    }
+
+    /// @notice  After timelock is over unlock KP3R and transfer to treasury
+    function unlockKP3R() external onlyGuardian {
+        KP3RVault.withdraw();
+
+        uint amount = KP3R.balanceOf(address(this));
+
+        KP3R.safeTransfer(address(treasury), amount);
+    }
+
+    /// @notice         Transfers specified ERC20 and amount to treaury
+    /// @param _asset   Addres of asset to transfer to treasury
+    /// @param _amount  Amount of `_asset` to be transfered to treasury
+    function withdraw(IERC20 _asset, uint256 _amount) external onlyGuardian {
+        _asset.safeTransfer(address(treasury), _amount);
+    }
+
+    /// @notice            Vote in Fixed Forex Gauge
+    /// @param _tokenVote  Addresses of tokens that are being voted on
+    /// @param _weights    Weight of vote each corresponding token recieves
+    function vote(address[] calldata _tokenVote, uint[] calldata _weights) external onlyGuardian {
+        gauge.vote(_tokenVote, _weights);
+    }
+
+    /// @notice  Claim rKP3R
+    function rKP3RClaim() external {
+        distributor.claim();
+    }
+
+    /// @notice  Burn rKP3R and mint oKP3R
+    function oKP3Rclaim() external onlyGuardian {
+        rKP3R.claim();
+    }
+
+    /// @notice         Exercise oKP3R, pay in USDC
+    /// @param _id      ID of option to exercise / oKP3R to burn
+    /// @param _manage  Bool if managing USDC from treasury
```

Comment:
Comment is wrong, should be KP3R i think

Reply:
Is right I believe. https://etherscan.io/address/0xEdB67Ee1B171c4eC66E6c10EC43EDBbA20FaE8e9#code

L1007